client/src/Tools/_framework/ToolPanels/MoveContentToFolder.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useState } from "react";
+import {
+  Text,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  SimpleGrid,
+  Heading,
+  IconButton,
+  HStack,
+} from "@chakra-ui/react";
+import axios from "axios";
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { useFetcher } from "react-router-dom";
+import { ActivityStructure } from "../Paths/ActivityEditor";
+
+interface activeView {
+  folder: null | {
+    name: string;
+    id: number;
+    parentFolderId: number | null;
+  };
+  contents: {
+    isFolder: boolean;
+    name: string;
+    id: number;
+  }[];
+}
+
+export default function MoveContentToFolder({
+  isOpen,
+  onClose,
+  id,
+  currentParentId,
+}) {
+  // Set when the modal opens
+  const [parentId, setParentId] = useState<number | null>(null);
+  const [contentName, setContentName] = useState<string>("");
+  useEffect(() => {
+    if (isOpen) {
+      // TODO: does this async function needed to be awaited for before we end this effect?
+      updateActiveView(currentParentId, true);
+    }
+  }, [isOpen]);
+
+  // Set whenever the user navigates to another folder
+  const [activeView, setActiveView] = useState<activeView>({
+    folder: null,
+    contents: [],
+  });
+  async function updateActiveView(
+    newActiveFolderId: number | null,
+    modalJustOpened: boolean = false,
+  ) {
+    const { data } = await axios.get(
+      `/api/getMyFolderContent/${newActiveFolderId ?? ""}`,
+    );
+
+    const folderName: string = data.folder?.name ?? null;
+    const parentFolderId: number = data.folder?.parentFolder?.id ?? null;
+    const contentFromApi: ActivityStructure[] = data.content;
+
+    const content = contentFromApi
+      .map((item) => {
+        return {
+          isFolder: Boolean(item.isFolder),
+          name: item.name,
+          id: item.id,
+        };
+      })
+      .sort((a, b) => {
+        if (a.isFolder && !b.isFolder) {
+          return -1;
+        } else if (b.isFolder && !a.isFolder) {
+          return 1;
+        } else {
+          return 0;
+        }
+      });
+
+    const folderInfo =
+      newActiveFolderId === null
+        ? null
+        : {
+            name: folderName,
+            id: newActiveFolderId,
+            parentFolderId,
+          };
+
+    setActiveView((_) => {
+      return { folder: folderInfo, contents: content };
+    });
+
+    if (modalJustOpened) {
+      const { name: movableContentName } = content.find(
+        (item) => item.id === id,
+      )!;
+      setContentName((_) => movableContentName);
+      setParentId((_) => currentParentId);
+    }
+  }
+
+  const initialRef = React.useRef(null);
```

Comment:
Add type or it thinks it is `null`:
```
const initialRef = React.useRef<HTMLButtonElement>(null);
```
(It's funny how you don't put `| null` in the the type, I guess because it is immutable and doesn't get the value until it is assigned by being in the `ref` attribute.)